src/txdb.cpp

```diff
@@ -168,6 +170,17 @@ bool CBlockTreeDB::ReadLastBlockFile(int &nFile) {
     return Read(DB_LAST_BLOCK, nFile);
 }
 
+bool CBlockTreeDB::ReadXFieldAggpubkeys(std::vector<XFieldEntry> & xFieldList) {
+    return Read(DB_XFIELD_AGGPUBKEY, xFieldList);
+}
+
+bool CBlockTreeDB::AddXFieldAggpubkey(XFieldEntry & xFieldEntry) {
+    std::vector<XFieldEntry> xFieldList;
+    Read(DB_XFIELD_AGGPUBKEY, xFieldList);
+    xFieldList.push_back(xFieldEntry);
```

Comment:
There is a check before this call in `validation.cpp` 

`(CPubKey(...) != FederationParams().GetLatestAggregatePubkey()))`

This check verifies that the new aggregate pub key is not the same as the previous one. But older aggregate pub keys may be accepted again.

This scenario is also covered by the functional test where the first two aggregate pub keys are repeated
```
expectedAggPubKeys = [
            { self.aggpubkeys[0] : 0},
            { self.aggpubkeys[1] : 12},
            { self.aggpubkeys[2] : 25},
            { self.aggpubkeys[3] : 33},
            { self.aggpubkeys[4] : 37},
            { self.aggpubkeys[0] : 42},
            { self.aggpubkeys[1] : 43}, ]

Reply:
I see, but `XFieldEntry` has block height and hash in addition to aggregated public key. therefore this value do not duplicate. and it would be better to make this check a feature of this method, so that the necessary checks are not dispersed at the time of the call.